mscore/noteGroups.cpp

```diff
@@ -148,10 +152,31 @@ void NoteGroups::resetClicked()
 void NoteGroups::noteClicked(Note* note)
       {
       Chord* chord = note->chord();
-      if (chord->beamMode() == Beam::Mode::AUTO)
-            chord->setBeamMode(Beam::Mode::BEGIN);
-      else if (chord->beamMode() == Beam::Mode::BEGIN)
+
+      Ms::Beam::Mode selectedBeamMode;
+      switch (iconPalette->getSelectedIdx()) {
+            case 0:
+                  selectedBeamMode = Beam::Mode::BEGIN;
+                  break;
+            case 1:
+                  selectedBeamMode = Beam::Mode::MID;
+                  break;
+            case 2:
+                  selectedBeamMode = Beam::Mode::BEGIN32;
+                  break;
+            case 3:
+                  selectedBeamMode = Beam::Mode::BEGIN64;
+                  break;
+            default:
+                  selectedBeamMode = Beam::Mode::BEGIN;
+                  break;
+            }
+
+      if (chord->beamMode() != selectedBeamMode)
+            chord->setBeamMode(selectedBeamMode);
+      else
             chord->setBeamMode(Beam::Mode::AUTO);
+
```

Comment:
note that behavior of above if/else is slightly different than was previously, but I much prefer my new behavior.  With the old version, clicking on a note that wasn't either AUTO or BEGIN would do nothing, so I'm afraid user would be confused and think that the UI was simply not working.  With my new behavior, clicking on a note will always do something to the beam mode.